### PR TITLE
cleanup around sending all "terse update" data whenever some of it changes

### PR DESCRIPTION
**Pull request by AndrewMeadows**
This is some cleanup that happened during an exploration into reducing some of the glitches seen in the simulation of "observing" interfaces.  The glitches aren't solved yet (we're going to punt them for a little while) but I've added some comments to help me resume the effort later.

### CODE REVIEW

File: libraries/entities/src/EntityItem.cpp [hunk 1]
```diff
@@ -944,31 +944,38 @@ void EntityItem::getAllTerseUpdateProperties(EntityItemProperties& properties) c
...
     SET_ENTITY_PROPERTY_FROM_PROPERTIES(glowLevel, setGlowLevel);
     SET_ENTITY_PROPERTY_FROM_PROPERTIES(localRenderAlpha, setLocalRenderAlpha);
     SET_ENTITY_PROPERTY_FROM_PROPERTIES(visible, setVisible);
-    SET_ENTITY_PROPERTY_FROM_PROPERTIES(ignoreForCollisions, updateIgnoreForCollisions);
-    SET_ENTITY_PROPERTY_FROM_PROPERTIES(collisionsWillMove, updateCollisionsWillMove);
     SET_ENTITY_PROPERTY_FROM_PROPERTIES(locked, setLocked);
     SET_ENTITY_PROPERTY_FROM_PROPERTIES(userData, setUserData);
-    SET_ENTITY_PROPERTY_FROM_PROPERTIES(simulatorID, updateSimulatorID);
     SET_ENTITY_PROPERTY_FROM_PROPERTIES(marketplaceID, setMarketplaceID);
     SET_ENTITY_PROPERTY_FROM_PROPERTIES(name, setName);
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityItem.cpp]: I've got an idea on how to better keep track of **what exactly** changed when an entity's properties changes, but I'm not implementing the full change here.  I'm just sorting the properties changes to make it easier later.

File: libraries/entities/src/EntityScriptingInterface.cpp [hunk 2]
```diff
@@ -128,37 +128,45 @@ EntityItemProperties EntityScriptingInterface::getEntityProperties(QUuid identit
...
     if (_entityTree) {
         _entityTree->lockForWrite();
-        _entityTree->updateEntity(entityID, properties);
+        bool updatedEntity = _entityTree->updateEntity(entityID, properties);
         _entityTree->unlock();
-    }
 
-    // make sure the properties has a type, so that the encode can know which properties to include
-    if (properties.getType() == EntityTypes::Unknown) {
-        EntityItemPointer entity = _entityTree->findEntityByEntityItemID(entityID);
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityScriptingInterface.cpp]: We used to access _entityTree above, even if it happened to be NULL.  We weren't crashing here yet (it was never NULL) but I'm cleaning up the logic to be safe.

File: libraries/entities/src/EntityScriptingInterface.cpp [hunk 3]
```diff
@@ -128,37 +128,45 @@ EntityItemProperties EntityScriptingInterface::getEntityProperties(QUuid identit
...
             }
-            queueEntityMessage(PacketTypeEntityEdit, entityID, modifiedProperties);
+            _entityTree->unlock();
+            queueEntityMessage(PacketTypeEntityEdit, entityID, properties);
             return id;
         }
+        return QUuid();
     }
 
     queueEntityMessage(PacketTypeEntityEdit, entityID, properties);
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityScriptingInterface.cpp]: This code relates to some of the problem with the glitches we're seeing.  Whenever a script changes any property we immediately send an update to the entity-server even if the property would normally fall to the physics simulation.  The solution will be to NOT send such updates and instead let the physics simulation send its result.

File: libraries/physics/src/EntityMotionState.cpp [hunk 4]
```diff
@@ -49,24 +49,17 @@ EntityMotionState::~EntityMotionState() {
...
-    if (flags & EntityItem::DIRTY_ANGULAR_VELOCITY) {
-        _serverAngularVelocity = _entity->getAngularVelocity();
-    }
+void EntityMotionState::updateServerPhysicsVariables() {
+    _serverPosition = _entity->getPosition();
+    _serverRotation = _entity->getRotation();
+    _serverVelocity = _entity->getVelocity();
+    _serverAngularVelocity = _entity->getAngularVelocity();
+    _serverAcceleration = _entity->getAcceleration();
 }
```
Comment (AndrewMeadows) [inline libraries/physics/src/EntityMotionState.cpp]: Whenever we send any "terse update" data to the entity-server we need to update ALL of the terse update properties in our local prediction of the entity-server's simulation.

**Comment by hifi-gustavo** [comment]
Build console output for commit: acb3163f4306c59d5f4aba74897149cea87ce1d6 available at: https://gist.github.com/75f39f66ea9f6c474b73<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5027.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5027.dmg

**Pull request merged**